python/lsst/ts/fbs/utils/maintel/make_fieldsurvey_scheduler.py

```diff
@@ -0,0 +1,147 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MakeFieldSurveyScheduler"]
+
+import typing
+
+from rubin_scheduler.scheduler.basis_functions import BaseBasisFunction
+from rubin_scheduler.scheduler.schedulers import CoreScheduler
+from rubin_scheduler.scheduler.surveys import BaseSurvey, FieldSurvey
+
+from ..data import field_survey_centers
+
+
+class MakeFieldSurveyScheduler:
+    """Construct Feature Based Scheduler configuration consisting of an
+    ensemble of field surveys, and optionally, other types of surveys.
+    """
+
+    def __init__(
+        self,
+        nside: int = 32,
+        ntiers: int = 1,
+    ) -> None:
+
+        self.nside = 32
+        self.surveys: typing.List[typing.List[BaseSurvey]] = [
+            [] for _ in range(0, ntiers)
+        ]
+
+    def _load_candidate_fields(self) -> typing.Dict:
+        """Load pointing center data for field surveys."""
+        name_fields, ra_fields, dec_fields = zip(*field_survey_centers.fields)
+        fields = {}
+        for name, ra, dec in zip(name_fields, ra_fields, dec_fields):
+            fields[name] = {"RA": ra, "dec": dec}
+        return fields
+
+    def add_field_surveys(
+        self,
+        tier: int,
+        json_block: str,
+        observation_reason: str,
+        science_program: str,
+        field_names: typing.List[str],
+        basis_functions: typing.List[BaseBasisFunction] = [],
+        **kwargs: typing.Any,
+    ) -> None:
+        """Add a list of field surveys to the scheduler configuration.
+
+        Parameters
+        ----------
+        tier: `int`
+            Tier index used to control prioritization of surveys.
+        json_block: `str`
+            JSON BLOCK used to perform the observations.
+        observation_reason: `str`
+            Purpose of the observation, e.g., "science"
+        science_program: `str`
+            Name of the science program.
+        field_names: `list` of `str`
+            List of names to specify the pointing center of each field survey.
+        basis_functions: `list` of `BaseBasisFunction`
+            Basis functions provided to each field survey.
+        """
+
+        if "survey_name" in kwargs:
+            raise ValueError("Use program_name rather than survey_name.")
+
+        fields = self._load_candidate_fields()
+
+        for field_name in field_names:
+            field_ra_deg = fields[field_name]["RA"]
+            field_dec_deg = fields[field_name]["dec"]
+
+            self.surveys[tier].append(
+                FieldSurvey(
+                    basis_functions,
+                    field_ra_deg,
+                    field_dec_deg,
+                    # sequence=sequence,
+                    # nvisits=nvisits,
+                    # exptimes=exptimes,
+                    # nexps=nexps,
+                    ignore_obs=None,
+                    survey_name=None,
```

Comment:
We should use a configurable survey_name here.

Reply:
Revising this comment to say we should use the science_program as the survey_name here to match nextVisit metadata output.